products/appengine/api.yaml

```diff
@@ -459,6 +459,800 @@ objects:
+                  required: true
+                - !ruby/object:Api::Type::String
+                  name: 'volumeType'
+                  description: |
+                    Underlying volume type, e.g. 'tmpfs'.
+                  required: true
+                - !ruby/object:Api::Type::Integer
+                  name: 'sizeGb'
+                  description: |
+                    Volume size in gigabytes.
+                  required: true
+      - !ruby/object:Api::Type::String
+        name: 'runtime'
+        description: |
+          Desired runtime. Example python27.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'runtimeChannel'
+        description: |
+          The channel of the runtime to use. Only available for some runtimes.
+      - !ruby/object:Api::Type::Boolean
+        name: 'threadsafe'
+        description: |
+          Whether multiple requests can be dispatched to this version at once.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'betaSettings'
+        description: |
+          Metadata settings that are supplied to this version to enable beta runtime features.
+      - !ruby/object:Api::Type::Enum
+        name: 'servingStatus'
+        description: |
+          Current serving status of this version. Only the versions with a SERVING status create instances and can be billed.
+          Defaults to SERVING.
+        default_value: :SERVING
+        values:
+          - :SERVING
+          - :STOPPED
+      - !ruby/object:Api::Type::String
+        name: 'runtimeApiVersion'
+        description: |
+          The version of the API in the given runtime environment. 
+          Please see the app.yaml reference for valid values at https://cloud.google.com/appengine/docs/standard//config/appref
+      - !ruby/object:Api::Type::String
+        name: 'runtimeMainExecutablePath'
+        description: |
+          The path or name of the app's main executable.
+      - !ruby/object:Api::Type::Array
+        name: 'handlers'
+        description: |
+          An ordered list of URL-matching patterns that should be applied to incoming requests. 
+          The first matching URL handles the request and other request handlers are not attempted.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+          - !ruby/object:Api::Type::String
+            name: 'urlRegex'
+            required: true
+            description: |
+              URL prefix. Uses regular expression syntax, which means regexp special characters must be escaped, but should not contain groupings. 
+              All URLs that begin with this prefix are handled by this handler, using the portion of the URL after the prefix as part of the file path.
+          - !ruby/object:Api::Type::Enum
+            name: 'securityLevel'
+            description: |
+              Security (HTTPS) enforcement for this URL.
+            values:
+              - :SECURE_UNSPECIFIED
```

Comment:
A lot of these enums have a 0 value (default value) of `SOMETHING_UNSPECIFIED`. But the API almost always rejects a request that contains one of these values. AFAIK it's primarily used for internal error handling of uninitialized objects. 99% of the time we should not expose these values to Terraform users.